vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -606,6 +610,12 @@ def check(self, _):
         # type: (Any) -> None
         self._hostname = datadog_agent.get_hostname()
         # Assert the health of the vCenter API by getting the version, and submit the service_check accordingly
+
+        now = get_timestamp()
+        if self.last_connection_time + self._config.connection_reset_timeout <= now or self.api is None:
+            self.last_connection_time = now
+            self.initiate_api_connection()
+
         try:
```

Comment:
We should have a test for this, at least testing different values of `last_connection_time` and one where the api is refreshed when it is previously None